src/content/1.7/modules/sample-modules/extending-sf-form-with-upload-image-field.md

```diff
@@ -0,0 +1,586 @@
+---
+title: Extending Symfony form with upload image field
+weight: 3
+aliases: 
+    - /1.7/modules/sample_modules/grid-and-identifiable-object-form-hooks-usage
+---
+
+# Extending Symfony form with upload image field
+{{< minver v="1.7.7" title="true" >}}
+
+
+## Introduction
+
+In this tutorial we are going to build a module which extends `Suppliers` form 
+(SELL -> Catalog -> Brands & Suppliers). You will learn how to create:
+
+- Main module class
+- Installer class
+- Create Symfony controller
+- Doctrine entity
+- Repository class
+- Image Uploader class
+- Twig View template
+
```

Comment:
I suggest we explain quickly what each class will do, that gives context to the reader.
```
- Main module class
- Installer class
- Create Symfony controller
- Doctrine entity
- Repository class
- Image Uploader class
- Twig View template
```
⬇️ 
```
- Main module class: main module entry point and hook entry point
- Installer class: responsible for module installation and uninstallation process
- Create Symfony controller: needed as we add a new "delete image" controller action
- Doctrine entity: this model is responsible for the image data persistence
- Repository class: this model is for image database search and retrieve
- Image Uploader class: this class is responsible for image upload process
- Twig View template: needed for display
```